### PR TITLE
Fix bug in AstroCalc/Eclipses where type of local solar eclipses are incorrect

### PR DESCRIPTION
**Pull request by worachate001**
### Description
There is a bug in the AstroCalc's Local Solar Eclipses tool (in development) that must be fixed.

To reproduce
* open 'All Solar Eclipses' tab
* search for eclipses in the year 1986
* double click at 1986 Oct 3
* go to 'Local Solar Eclipses' tab
* search for eclipses in the same year

The result shows that 1986 Oct 3 was a partial eclipse as seen from the location of greatest eclipse along very narrow eclipse path (path width = 0.46 km) which is incorrect. For (unrealistically) perfect sphere Moon, this eclipse was a total eclipse with sub-second duration - 0.049 second - to be exact. In reality, it's beaded total eclipse. Both NASA web site and Stellarium tell us that it's hybrid eclipse - technically, it was total eclipse at mid point.

This PR will correctly classify this rare event as total eclipse as seen from the location and it will also fix similar situation on other dates when you are standing at the point of transition between total and annular eclipse.

Notes:
* This PR will fix the oblateness parameters of the Earth to be consistence across all related parts of Stellarium.
* As we reproduce the bug, we can see that it seems to be a slight difference of Sun & Moon positions at greatest eclipse point for some eclipses. This is still under investigations, would be addressed in a separate issue if I found a way to fix it.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:
* Operating system: Windows 10
* Graphics Card: Nvidia

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by alex-w — APPROVED** [review]
Thanks!

File: src/gui/AstroCalcDialog.cpp [hunk 1]
```diff
@@ -3018,11 +3018,15 @@ pSEparameter partialSolarEclipse() {
 	double L1 = bessel.L1;
 	double L2 = bessel.L2;
 	const double mu = bessel.mu;
-	constexpr double e2 = 0.00669398;
-	// e^2 = 0.00669398 : Earth flattening parameter
-	// f = 298.25642 : e^2 = 2f-f^2
-	// Source: IERS Conventions (2003)
+	constexpr double e2 = 0.00669438;
+	// e^2 = 0.00669438 : Earth flattening parameter
+	// Inverse flattening 1/f = 298.257223563 : e^2 = 2f-f^2
```
Comment (gzotti) [inline src/gui/AstroCalcDialog.cpp:3023]: We are actually using this as property of the Planet object Earth. See ssystem_major.ini entry for Earth: 
`oblateness=0.003352810664747481
`
This is f. 
The parameter is used in Planet::getOneMinusOblateness() and therefore in Planet::getRectangularCoordinates(), which was crucial to get ... accurate eclipse simulation for 2017. 

For consistency I would prefer if you don't hardcode the value here but use Planet::getOneMinusOblateness(), should we ever want to change to the IERS2003 or other values. But please keep the documentation, this is important for traceability.

**Comment by gzotti** [comment]
@worachate001 Thank you for your meticulous work! It's fine to have contributors who triple-check their work, even for such incredibly rare cases :-)

File: src/gui/AstroCalcDialog.cpp (hunk 1 above)
Comment (worachate001) [reply src/gui/AstroCalcDialog.cpp:3023]: Thanks, I'll add it in my to do list.

**Pull request merged**